department-of-veterans-affairs/caseflow
# Handle viewing attorney's cases outside their team

Resolves #12999 

### Description
Show error message when judge attempt to view the assigned cases of an attorney outside their team.

### Acceptance Criteria
- [x] Code compiles correctly and tests pass

### Testing Plan
In your browser, enable DevTools > Network to scan for error status codes.
1. Log in as a judge
1. Go to their Case Assign page (e.g., http://localhost:3000/queue/BVAAABSHIRE/assign)
1. Click on attorneys on the left sidepanel to see their assigned cases; note the URLs.
1. Go to http://localhost:3000/queue/BVAAABSHIRE/assign/92 or other attorney `user_id`s (who are not on the judge's team, such as `JudgeTeam.for_judge(User.find_by(css_id: "BVAGSPORER")).non_admins`). An "Attorney is not part of the specified judge's team" error page should appear as shown in screenshot below.
    1. Navigate back via browser; confirm the error does not persist.
    1. Retrigger the error and then select a valid attorney queue and confirm the error does not persist.
1. Log in as BVAGSPORER, and go to one of BVAAABSHIRE's attorney's assigned cases, such as http://localhost:3000/queue/BVAAABSHIRE/assign/88. Should get an "Additional access needed" error page (due to the 403 Forbidden return status from the `TasksController`).

Now enable SCM (`FeatureToggle.enable!(:scm_view_judge_assign_queue)`) and log in as an SCM user (e.g., BVARDUNKLE), who should be able to go to any judge's case assign page and their attorneys' case assignments page without any error.

### User Facing Changes

 BEFORE|AFTER
 ---|---
<img width="722" alt="image" src="https://user-images.githubusercontent.com/55255674/81072228-77484c00-8eab-11ea-9ac3-87d7a17176cb.png"> (blank screen; unhandled error in devtools) | <img width="1159" alt="image" src="https://user-images.githubusercontent.com/55255674/81129474-07b57980-8f0a-11ea-81f1-250f71d91ebd.png"> (Atty 92 is not part of BVAAABSHIRE's JudgeTeam), <img width="994" alt="image" src="https://user-images.githubusercontent.com/55255674/81129529-3df2f900-8f0a-11ea-8ba2-f853018e6bdc.png"> (Atty 92 is part of BVAGSPORER's JudgeTeam, which BVAAABSHIRE is not a part of)




### Code Documentation Updates
